dataprofiler/profilers/profile_builder.py

```diff
@@ -309,7 +325,8 @@ def get_base_props_and_clean_null_params(self, df_series, sample_size,
                 break
             
         # close the generator in case it is not exhausted.
-        sample_ind_generator.close()
+        if sample_ids is None:
```

Comment:
Why is this checking if `sample_ids` is None and not if `sample_ind_generator` is None?

Reply:
Furthermore why is any check necessary since either way a sample_ind_generator is created?